### PR TITLE
fix: Fix Range intersects algorithm

### PR DESCRIPTION
**Pull request by max-b**
Fixes https://github.com/npm/node-semver/issues/239

I think there's still an open question about whether `<1.0.0 >2.0.0` and `>0.0.0` intersect because technically `<1.0.0 >2.0.0` is an empty set and so can't intersect with anything, but that would require a more complicated solution and this PR is a more trivial fix.

### CODE REVIEW

**Comment by isaacs** [comment]
Yeah, `<1.0.0 >2.0.0` should not intersect `>0.0.0`.

Intersect should not be "is the first set unchanged by adding the second set", it's "is there a version somewhere that satisfies both ranges".

**Comment by max-b** [comment]
Ok I think that could be addressed with something like:
```
  return this.set.some(function (thisComparators) {
    return range.set.some(function (rangeComparators) {
      return isSatisfiable(thisComparators) && thisComparators.every(function (thisComparator) {
        return isSatisfiable(rangeComparators) && rangeComparators.every(function (rangeComparator) {
          return thisComparator.intersects(rangeComparator, options)
        })
      })
    })
  })
```
Where `isSatisfiable` would do essentially what it says on the lid.

Would you consider this a breaking change? It's technically a fix, but it will require changing test results and it's possible that there's someone out there depending on the previous behavior/idiosyncrasy?

**Comment by max-b** [comment]
Great! I've been using semantic-release style commits out of habit, but it doesn't really look like they're being fully used here (and I think none of them would actually trigger a major version change...). I'd imagine that if everything looks ok you can decide how'd you like to merge/rebase it.

Thanks!

File: semver.js [hunk 1]
```diff
@@ -799,11 +799,15 @@ Range.prototype.intersects = function (range, options) {
 
   return this.set.some(function (thisComparators) {
     return range.set.some(function (rangeComparators) {
-      return isSatisfiable(thisComparators, options) && thisComparators.every(function (thisComparator) {
-        return isSatisfiable(rangeComparators, options) && rangeComparators.every(function (rangeComparator) {
-          return thisComparator.intersects(rangeComparator, options)
+      return (
+        isSatisfiable(thisComparators, options) &&
```
Comment (isaacs) [inline semver.js]: This could also be moved up a line.  No sense walking `range.set` if `thisComparators` isn't satisfiable.

**Comment by max-b** [comment]
Yay 🎉 Thanks for the feedback/hand-holding 😬 

Should be able to close #239 now 😄

**Comment by isaacs** [comment]
This landed on v6.

**Pull request merged**